resource-validator/src/main/scala/com/broadinstitute/dsp/resourceValidator/DbReader.scala

```diff
@@ -38,5 +41,30 @@ object DbReader {
         .query[Runtime]
         .stream
         .transact(xa)
+
+    override def getBucketsToDelete: Stream[F, BucketToRemove] =
+      sql"""select googleProject, stagingBucket from CLUSTER WHERE status="Deleted" and destroyedDate < now() - interval 15 DAY;"""
+        .query[BucketToRemove]
+        .stream
+        .transact(xa)
+
+    // Return all clusters that has no non "deleted" nodepool
+    override def getK8sClustersToDelete: Stream[F, KubernetesClusterId] =
+      sql"""SELECT kc.id, kc.location, kc.clusterName
+            FROM KUBERNETES_CLUSTER kc 
+            WHERE NOT EXISTS (
+              SELECT *
+                FROM KUBERNETES_CLUSTER kc 
+                LEFT JOIN NODEPOOL as np on np.clusterId=kc.id
+              WHERE np.status != "DELETED"
```

Comment:
May need to include `ERROR` as well to catch cases where nodepool deletion errored out or some other unexpected issues came up.

Reply:
it looks like nodepool can end up in `ERROR` state, so I don't think it should be here. I'll make another checker similar to https://github.com/broadinstitute/leonardo-cron-jobs/blob/master/resource-validator/src/main/scala/com/broadinstitute/dsp/resourceValidator/ErroredRuntimeChecker.scala

Reply:
This feels a bit different than what `ErroredRuntimeChecker` does because we're interested not only in cleaning up errored nodepools but also the cluster they belong to, if there are no non-deleted and non-errored nodepools on that cluster. Also happy to see how you want to implement it differently, though.

Reply:
It's similar to `ErroredRuntimeChecker` only in a sense that one cares about errored runtime, and one cares about errored nodepool...The action taken will be different in many regards as well, the only similar part is if we detect `ERROR` in google for nodepool, we don't want to delete k8s cluster (similar to we don't want to delete errored runtime)